### ISSUE TITLE
Command failed: odo list

### ISSUE THREAD

**Issue by mhagnumdw** [redhat-developer/vscode-openshift-tools#1824]
### Versions
- Fedora 32 x86_64
- uname -r: 5.8.7-200.fc32.x86_64
- vscode 1.50.1
- Extension version: 0.2.0
- OpenShift 4.5.2

### Log

```log
Command failed: "/home/USER/.vscode/extensions/redhat.vscode-openshift-connector-0.2.0/out/tools/linux/odo" list --app my-app --project my-project -o json
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x50 pc=0x166add2]

goroutine 1 [running]:
github.com/openshift/odo/pkg/component.getRemoteComponentMetadata(0xc000155d90, 0xc000422d60, 0x20, 0x7fff78cdab71, 0xa, 0x7fff78cdab86, 0x6, 0x101, 0x0, 0x0, ...)
	/builddir/build/BUILD/gocode/src/github.com/openshift/odo/pkg/component/component.go:1414 +0xd2
github.com/openshift/odo/pkg/component.GetComponent(...)
	/builddir/build/BUILD/gocode/src/github.com/openshift/odo/pkg/component/component.go:1403
github.com/openshift/odo/pkg/component.List(0xc000155d90, 0x7fff78cdab71, 0xa, 0xc0002535e0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/builddir/build/BUILD/gocode/src/github.com/openshift/odo/pkg/component/component.go:882 +0x90a
github.com/openshift/odo/pkg/odo/cli/component.(*ListOptions).Run(0xc00082a9c0, 0x0, 0x0)
	/builddir/build/BUILD/gocode/src/github.com/openshift/odo/pkg/odo/cli/component/list.go:273 +0x15e8
github.com/openshift/odo/pkg/odo/genericclioptions.GenericRun(0x2227fc0, 0xc00082a9c0, 0xc0004be000, 0xc00093ff80, 0x0, 0x6)
	/builddir/build/BUILD/gocode/src/github.com/openshift/odo/pkg/odo/genericclioptions/runnable.go:31 +0x13c
github.com/openshift/odo/pkg/odo/cli/component.NewCmdList.func1(0xc0004be000, 0xc00093ff80, 0x0, 0x6)
	/builddir/build/BUILD/gocode/src/github.com/openshift/odo/pkg/odo/cli/component/list.go:338 +0x5e
github.com/spf13/cobra.(*Command).execute(0xc0004be000, 0xc00093ff20, 0x6, 0x6, 0xc0004be000, 0xc00093ff20)
	/builddir/build/BUILD/gocode/src/github.com/openshift/odo/vendor/github.com/spf13/cobra/command.go:830 +0x2aa
github.com/spf13/cobra.(*Command).ExecuteC(0xc000590c80, 0x1fd50c8, 0xc000061740, 0x7)
	/builddir/build/BUILD/gocode/src/github.com/openshift/odo/vendor/github.com/spf13/cobra/command.go:914 +0x2fb
github.com/spf13/cobra.(*Command).Execute(...)
	/builddir/build/BUILD/gocode/src/github.com/openshift/odo/vendor/github.com/spf13/cobra/command.go:864
main.main()
	/builddir/build/BUILD/gocode/src/github.com/openshift/odo/cmd/odo/odo.go:67 +0x330

```

**Comment by mohitsuman** [comment]
@mhagnumdw Can you please let us know how are you connecting to OpenShift instance using remote cluster or local OpenShift instance ?

**Comment by mhagnumdw** [comment]
> @mhagnumdw Can you please let us know how are you connecting to OpenShift instance using remote cluster or local OpenShift instance ?

Remote cluster.

**Comment by dgolovin** [comment]
@mhagnumdw is there any output for `OpenShift: About` when running from command palette?

**Comment by mhagnumdw** [comment]
> @mhagnumdw is there any output for `OpenShift: About` when running from command palette?

I will only have access to the machine from my first post next Tuesday, 11/3.

But I installed the extension in my current environment and got the same problem, following information:

### Versions
- 20.04.1 LTS (Focal Fossa) x86_64 (via WSL2)
- uname -r: 4.19.104-microsoft-standard
- vscode 1.50.1
- Extension version: 0.2.0
- OpenShift 4.5.2 (Remote cluster)

### Log
```log
Command failed: "/home/mhagnumdw/.vscode-server/extensions/redhat.vscode-openshift-connector-0.2.0/out/tools/linux/odo" list --app my-app --project my-project -o json
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x50 pc=0x166add2]

goroutine 1 [running]:
github.com/openshift/odo/pkg/component.getRemoteComponentMetadata(0xc00072f290, 0xc00055e840, 0x20, 0x7ffea9778db4, 0xa, 0x7ffea9778dc9, 0x6, 0x101, 0x0, 0x0, ...)
	/builddir/build/BUILD/gocode/src/github.com/openshift/odo/pkg/component/component.go:1414 +0xd2
github.com/openshift/odo/pkg/component.GetComponent(...)
	/builddir/build/BUILD/gocode/src/github.com/openshift/odo/pkg/component/component.go:1403
github.com/openshift/odo/pkg/component.List(0xc00072f290, 0x7ffea9778db4, 0xa, 0xc0008c76c0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/builddir/build/BUILD/gocode/src/github.com/openshift/odo/pkg/component/component.go:882 +0x90a
github.com/openshift/odo/pkg/odo/cli/component.(*ListOptions).Run(0xc0006e2120, 0x0, 0x0)
	/builddir/build/BUILD/gocode/src/github.com/openshift/odo/pkg/odo/cli/component/list.go:273 +0x15e8
github.com/openshift/odo/pkg/odo/genericclioptions.GenericRun(0x2227fc0, 0xc0006e2120, 0xc000428500, 0xc0008fc300, 0x0, 0x6)
	/builddir/build/BUILD/gocode/src/github.com/openshift/odo/pkg/odo/genericclioptions/runnable.go:31 +0x13c
github.com/openshift/odo/pkg/odo/cli/component.NewCmdList.func1(0xc000428500, 0xc0008fc300, 0x0, 0x6)
	/builddir/build/BUILD/gocode/src/github.com/openshift/odo/pkg/odo/cli/component/list.go:338 +0x5e
github.com/spf13/cobra.(*Command).execute(0xc000428500, 0xc0008fc2a0, 0x6, 0x6, 0xc000428500, 0xc0008fc2a0)
	/builddir/build/BUILD/gocode/src/github.com/openshift/odo/vendor/github.com/spf13/cobra/command.go:830 +0x2aa
github.com/spf13/cobra.(*Command).ExecuteC(0xc000582c80, 0x1fd50c8, 0xc0008e4780, 0x7)
	/builddir/build/BUILD/gocode/src/github.com/openshift/odo/vendor/github.com/spf13/cobra/command.go:914 +0x2fb
github.com/spf13/cobra.(*Command).Execute(...)
	/builddir/build/BUILD/gocode/src/github.com/openshift/odo/vendor/github.com/spf13/cobra/command.go:864
main.main()
	/builddir/build/BUILD/gocode/src/github.com/openshift/odo/cmd/odo/odo.go:67 +0x330
```

### OpenShift: About (from vscode command palette) - First time

```log
$ "/home/mhagnumdw/.vscode-server/extensions/redhat.vscode-openshift-connector-0.2.0/out/tools/linux/odo" version
odo v2.0.0 (6fbb9d9bf)

Server: https://api.dev.domain.com.br:6443
Kubernetes: v1.18.3+b74c5ed

---
A newer version of odo (v2.0.1) is available,
visit https://github.com/openshift/odo/releases to update.
If you wish to disable this notification, run:
odo preference set UpdateNotification false
---
```

### So I updated the odo, like this:
```bash
cd /home/mhagnumdw/.vscode-server/extensions/redhat.vscode-openshift-connector-0.2.0/out/tools/linux/
curl -L https://mirror.openshift.com/pub/openshift-v4/clients/odo/latest/odo-linux-amd64 -o odo
chmod +x /usr/local/bin/odo
# I followed instructions from here: https://github.com/openshift/odo/releases
```

### OpenShift: About (from vscode command palette) - Second time

```log
$ "/home/mhagnumdw/.vscode-server/extensions/redhat.vscode-openshift-connector-0.2.0/out/tools/linux/odo" version
odo v2.0.1 (a9fff7da4)

Server: https://api.dev.domain.com.br:6443
Kubernetes: v1.18.3+b74c5ed
```

### I tried using it again and again the error comes up

```log
Command failed: "/home/mhagnumdw/.vscode-server/extensions/redhat.vscode-openshift-connector-0.2.0/out/tools/linux/odo" list --app my-app --project my-project -o json
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x58 pc=0x167306f]

goroutine 1 [running]:
github.com/openshift/odo/pkg/component.getRemoteComponentMetadata(0xc000121a20, 0xc0007a1100, 0x20, 0x7ffce9866db4, 0xa, 0x7ffce9866dc9, 0x6, 0x101, 0x0, 0x0, ...)
	/builddir/build/BUILD/gocode/src/github.com/openshift/odo/pkg/component/component.go:1435 +0xcf
github.com/openshift/odo/pkg/component.GetComponent(...)
	/builddir/build/BUILD/gocode/src/github.com/openshift/odo/pkg/component/component.go:1424
github.com/openshift/odo/pkg/component.List(0xc000121a20, 0x7ffce9866db4, 0xa, 0xc0007a7a40, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/builddir/build/BUILD/gocode/src/github.com/openshift/odo/pkg/component/component.go:905 +0xb9b
github.com/openshift/odo/pkg/odo/cli/component.(*ListOptions).Run(0xc000828120, 0x0, 0x0)
	/builddir/build/BUILD/gocode/src/github.com/openshift/odo/pkg/odo/cli/component/list.go:272 +0x15e8
github.com/openshift/odo/pkg/odo/genericclioptions.GenericRun(0x2241de0, 0xc000828120, 0xc00043c780, 0xc00096ed80, 0x0, 0x6)
	/builddir/build/BUILD/gocode/src/github.com/openshift/odo/pkg/odo/genericclioptions/runnable.go:31 +0x13c
github.com/openshift/odo/pkg/odo/cli/component.NewCmdList.func1(0xc00043c780, 0xc00096ed80, 0x0, 0x6)
	/builddir/build/BUILD/gocode/src/github.com/openshift/odo/pkg/odo/cli/component/list.go:337 +0x5e
github.com/spf13/cobra.(*Command).execute(0xc00043c780, 0xc00096ed20, 0x6, 0x6, 0xc00043c780, 0xc00096ed20)
	/builddir/build/BUILD/gocode/src/github.com/openshift/odo/vendor/github.com/spf13/cobra/command.go:830 +0x2aa
github.com/spf13/cobra.(*Command).ExecuteC(0xc00043db80, 0x1fe9e80, 0xc00005f2c0, 0x7)
	/builddir/build/BUILD/gocode/src/github.com/openshift/odo/vendor/github.com/spf13/cobra/command.go:914 +0x2fb
github.com/spf13/cobra.(*Command).Execute(...)
	/builddir/build/BUILD/gocode/src/github.com/openshift/odo/vendor/github.com/spf13/cobra/command.go:864
main.main()
	/builddir/build/BUILD/gocode/src/github.com/openshift/odo/cmd/odo/odo.go:67 +0x330
```

### Print

![image](https://user-images.githubusercontent.com/17688846/97701965-fbf7d980-1a8c-11eb-8757-b77395307a22.png)

**Comment by dgolovin** [comment]
CC: @kadel FYI.

**Comment by dgolovin** [comment]
@mhagnumdw are you running Ubuntu out of Windows Store in Hyper-V?
Normally `uname -r` looks like
```
5.8.0-26-generic
```
For installation from original ISO.

**Comment by mhagnumdw** [comment]
> @mhagnumdw are you running Ubuntu out of Windows Store in Hyper-V?

@dgolovin I had installed version 18 from the Windows Store. And then I upgraded to version 20 by following the steps below:

```bash
sudo apt update && sudo apt upgrade
sudo do-release-upgrade
```

**Comment by dgolovin** [comment]
@mhagnumdw I cannot replicate it on latest original Ubuntu 20.10 running on `bare metal`, which is my laptop. Local or remote clusters works fine. 

Have you created any components using extension or you just tried to see structure of your app using Application Explorer View?
If you have any components deployed what did you use to create and deploy them? I would need that to replicate the error. Will try to deploy something with oc or devconsole to replicate.

**Comment by dgolovin** [comment]
@mhagnumdw I have replicated the problem, it seems that odo 2.0.0 detects app name from your deployments and then fails because none of those deployments created/managed by odo. 

Will follow open an issue and follow up with odo team.

**Comment by dgolovin** [comment]
Here is upstream issue for odo https://github.com/openshift/odo/issues/4183.

**Comment by drewhemm** [comment]
Hi, is there a workaround for this issue?

**Comment by kadel** [comment]
This got fixed upstream. So I guess this just needs to get into the release. Odo should have a new release this week.

**Comment by dgolovin** [comment]
Upstream issue is fixed and odo 2.0.4 is going to be included into next extension release.
